Update proposed charter to reflect BoF feedback (#1)
tls-attestation/expat-charter: README.md

# Background and Motivation

> In many scenarios, particularly with the rise of Trusted Execution Environments (TEEs) and the increasing security demands for IoT devices and confidential workloads, it's crucial also to ensure that the peer's state is as expected, in addition to conventional authentication (verification of identity).

Reviewer: "also" is not required here. It is covered by "in addition"

Author: Addressed in 8d4c6586c6e1133bc5ee868d86d6a00222985217

> In many scenarios, particularly with the rise of Trusted Execution Environments (TEEs) and the increasing security demands for IoT devices and confidential workloads, it's crucial also to ensure that the peer's state is as expected, in addition to conventional authentication (verification of identity).

Author: ```suggestion
In many scenarios, particularly with the rise of Trusted Execution Environments (TEEs) and the increasing security demands for IoT devices and confidential workloads, it's crucial to ensure that the peer's state is as expected, in addition to conventional authentication (verification of identity).
```

This provides a stronger assurance of trustworthiness, helping to prevent attacks that might compromise a system even if its traditional credentials remain valid, and enables authorization policies based on richer security signals.

To address this need for enhanced, verifiable trustworthiness, this WG will deliver a Standards Track solution using TLS.
The solution will bind to the TLS handshake and enable the attestation of one or both TLS endpoints.

This effort will focus on leveraging the (D)TLS handshake protocol as-is. It will also leverage the existing RATS toolbox to ensure interoperability with existing and future attestation technologies.

# Scope

This working group has a narrow focus: to produce a single document specifying an exchange that is performed after handshake completion and provides attestation of one or both TLS endpoints.

More specifically:
* The protocol should be agnostic to remote attestation technologies.

> * TLS 1.3 handshake protocol should be used as-is, leveraging its existing deployment and security guarantees, including formal validation.  In particular, authentication will only use existing TLS mechanisms.

Reviewer: Which authentication? It can be interpreted as post-handshake authentication. Propose: "authentication part of TLS handshake"

Reviewer 2: That sounds a bit awkward. How about...
```suggestion
* TLS 1.3 handshake protocol should be used as-is, leveraging its existing deployment and security guarantees, including formal validation.  In particular, authentication will only use the existing TLS handshake mechanisms.
```

> * The protocol may specify negotiation mechanisms within the initial TLS handshake provided they do not modify the cryptographic properties of the handshake.

Reviewer: This will be controversial. Let's remove it and do negotiation at application layer.

Author: It seems premature to close this door.  It’s a “may”; we’ll have the time to discuss this.

Reviewer: I heard people in the room saying remove number 3 in the slides.

Author: Yaron and I went through the minutes and chat to make sure all the feedback was accurately reflected.  We couldn’t find any explicit mention of it.  Can you provide a reference?

Reviewer 3: IMHO this is not a useful discussion. If Usama remembers that someone said something, that's good enough, no need for "a reference". On the other hand, we are ultimately accountable for this charter. So even if someone said something we can decide to reinterpret or even ignore it. It's just one person's opinion.

Reviewer: > Can you provide a reference?

Hannes in response to the concerns raised: [here](https://youtu.be/RS4LaqhupN4?t=6355).

Reviewer: [notes](https://notes.ietf.org/notes-ietf-123-expat)
> EKR: key resrictions are no change to TLS and to use post-handshake

Reviewer 4: I dont recollect ekr  insisting on "post-handshake" versus another TLS Extension that does not modify the key derivations of TLS.

Reviewer 5: I agree from my read of the room that I don't think the community has aligned on these requirements yet.  We may end up in a place where these hold true, but they shouldn't be starting-point restrictions.

> * The protocol must allow a Relying Party to ensure the freshness of the initial attestation, as well as conduct periodic re-attestations during the lifetime of the TLS connection.

Reviewer: re-attestation is very weird.

Author: What do you suggest instead?

Reviewer 2: ```suggestion
* The protocol must allow a Relying Party to ensure the freshness of the initial attestation and periodically refresh the attestation credentials throughout the lifetime of the TLS connection.
```

EDIT(@thomas-fossati): small tweaks.

Remote attestation ([RFC9334](https://datatracker.ietf.org/doc/rfc9334/)) addresses this by allowing an entity to produce verifiable Evidence about its current state - such as proving that its software and firmware haven't been tampered with, that secure boot is enabled, or that cryptographic keys are securely stored within a hardware-protected environment.

This provides a stronger assurance of trustworthiness, helping to prevent attacks that might compromise a system even if its traditional credentials remain valid, and enables authorization policies based on richer security signals.

To address this need for enhanced, verifiable trustworthiness, this WG will deliver a Standards Track solution using TLS.
The solution will bind to the TLS handshake and enable the attestation of one or both TLS endpoints.

This effort will focus on leveraging the (D)TLS handshake protocol as-is. It will also leverage the existing RATS toolbox to ensure interoperability with existing and future attestation technologies.

# Scope

> This working group has a narrow focus: to produce a single document specifying an exchange that is performed after handshake completion and provides attestation of one or both TLS endpoints.

Author: ```suggestion
This working group has a narrow focus: to produce a single document specifying an exchange that is performed after handshake completion and binds attestation of one or both peers to the TLS handshake.
```

> This working group has a narrow focus: to produce a single document specifying an exchange that is performed after handshake completion and binds attestation of one or both peers to the TLS handshake.

Reviewer 4: this is not correct. I think people are thinking of a TLS extension that completes along with the TLS handshake

Reviewer 5: I agree that my read of the room from the discussion is that this isn't necessarily agreed to yet and thus shouldn't be in the charter as a specific solution (see my note to the email list for longer text).

More specifically:
* The protocol should be agnostic to remote attestation technologies.
* TLS 1.3 handshake protocol should be used as-is, leveraging its existing deployment and security guarantees, including formal validation.  In particular, authentication will only use the existing TLS handshake mechanisms.
* The protocol may specify negotiation mechanisms within the initial TLS handshake provided they do not modify the cryptographic properties of the handshake.

> * The protocol must allow a Relying Party to ensure the freshness of the initial attestation and periodically refresh the attestation credentials throughout the lifetime of the TLS connection.

Author: ```suggestion
* The protocol must allow a Relying Party to ensure the [freshness](https://www.ietf.org/rfc/rfc9334.html#section-10) of the initial attestation and periodically refresh the attestation credentials throughout the lifetime of the TLS connection.
```

The working group will not extend TLS or create new remote attestation technologies or cryptographic protocol mechanisms.

> The working group will ensure that privacy principles and avoidance of vendor lock-ins are considered in the designed solution in line with the [IAB statement on Attestation](https://datatracker.ietf.org/doc/statement-iab-statement-on-the-risks-of-attestation-of-software-and-hardware-on-the-open-internet/).

Reviewer 4: I think the main concern was more, abusing attestation to refuse service, eg a rooted phone would not be able to connect to a gaming network via TLS.

The working group will engage with the research community on the formal analysis of the protocol artefacts in parallel with the specification work.

> The WG will consider using [Exported Authenticators (RFC9261)](https://datatracker.ietf.org/doc/rfc9261/) as the post-handshake mechanism and [Conceptual Message Wrapper (CMW)](https://datatracker.ietf.org/doc/draft-ietf-rats-msg-wrap/) as the format for the attestation credential. 

Reviewer 4: I think it would be better to leave this out of the charter. By saying "will consider" it is already sufficiently vague to not make an actual actionable statement. I also think a generic TLS extension that completes when the handshake completes is more likely to be picked than an authenticated exporter based solution.

Reviewer 5: FYI, that change was from what we presented on the screen and was incorporated here.  My goal in this was to not require the usage of a pre-existing solution, as that's not how the IETF works.  But it could be considered.  Given the discussion in the BOF I'd agree it should probably be removed entirely, as I don't know that community consensus exists beyond "it can be solved" and not toward a particular solution/direction.  I think the WG will need to do some heavy initial discussions around a best path forward and should not be restricted in its initial brain storming.

# Dependencies and Liaisons

> The working group will keep the TLS and RATS working groups, as well as the Confidential Computing Consortium (CCC) Attestation SIG informed of its activities by presenting updates at their meetings and requesting reviews.

Reviewer 4: I don't think liasioning to external organizations should be a WG responsibility listed in a charter? How would this be done and tracked by a WG? What if that organization has other meeting/informing requirements? That is why some of this goes via official liaison channels.

Reviewer: > I don't think liasioning to external organizations should be a WG responsibility listed in a charter?

I see from the chat that three different people (Ben, Ekr, Dave) have requested this. If you like, we can phrase it differently as in [WIMSE charter](https://datatracker.ietf.org/doc/charter-ietf-wimse/): 

"The expat working group will closely collaborate with:
- TLS and RATS working groups
- Confidential Computing Consortium (CCC) Attestation SIG."

> How would this be done and tracked by a WG? 

Interaction with the CCC is actually already happening. In fact, attested TLS is an _adopted_ project of [CCC Attestation SIG](https://github.com/CCC-Attestation). Also, @thomas-fossati is a chair at the SIG and I am regularly updating the SIG with the progress on the design and formal analysis in the meetings.

This provides a much stronger assurance of trustworthiness, helping to prevent attacks that might compromise a system even if its traditional credentials remain valid, and enables authorization policies based on richer security signals.

To address this need for enhanced, verifiable trustworthiness, this WG will deliver a Standards Track solution using TLS.

> The solution will bind to the TLS handshake and enable the attestation of one or both TLS endpoints.

Reviewer 2: Can we explicitly fold "binding" into (one of) the goals below? I thought that was one of the requests, but maybe I misunderstood the conclusion.

For example:
> This working group has a narrow focus: to produce a single document specifying an exchange that is performed after handshake completion and binds attestation of one or both peers to the TLS handshake.

Author: To be sure: @ionut-arm, are you suggesting to remove the new L9-10 and replace the new L16 with the text above?

Reviewer 2: No, I'm just suggesting a change of L16 so that we make the "binding" explicit in goals, 9-10 stays the same.

Author: Thanks for the clarification.  Addressed in 9365f225fbb507dfd7d4d65d15dc1cf76b0d8c8d

This effort will focus on leveraging the (D)TLS handshake protocol as-is. It will also leverage the existing RATS toolbox to ensure interoperability with existing and future attestation technologies.

# Goals

This working group has a narrow focus: to produce a single document specifying an exchange that is performed after handshake completion and provides attestation of one or both TLS endpoints.

More specifically:
* The protocol should be agnostic to remote attestation technologies.
* TLS 1.3 handshake protocol should be used as-is, leveraging its existing deployment and security guarantees, including formal validation.  In particular, authentication will only use existing TLS mechanisms.
* The protocol may specify negotiation mechanisms within the initial TLS handshake provided they do not modify the cryptographic properties of the handshake.

> * The protocol must allow a Relying Party to ensure that initial attested state is fresh, as well as periodic re-attestation during the lifetime of a TLS connection.

Reviewer 2: ```suggestion
* The protocol must allow a Relying Party to ensure the freshness of the initial attestation, as well as conduct periodic re-attestations during the lifetime of the TLS connection.
```

EDIT: slight rephrasing as per https://github.com/tls-attestation/expat-charter/pull/1#discussion_r2222050192

Author: I understand the overload you are trying to break, but I find the use of 'initial’ to be quite clear in this context.

However, on re-reading it, the sentence sounds a bit awkward, and I have a slight rephrasing suggestion:

> The protocol must allow a Relying Party to ensure the freshness of the initial attestation, as well as conduct periodic re-attestations during the lifetime of the TLS connection.

Reviewer 2: That's fine by me
